tests/components/climate/test_generic_thermostat.py

```diff
@@ -912,7 +995,321 @@ def _setup_sensor(self, temp, unit=TEMP_CELSIUS):
 
     def _setup_switch(self, is_on):
         """Setup the test switch."""
-        self.hass.states.set(ENT_SWITCH, STATE_ON if is_on else STATE_OFF)
+        self.hass.states.set(ENT_SWITCH_HEAT, STATE_ON if is_on else STATE_OFF)
+        self.calls = []
+
+        @callback
+        def log_call(call):
```

Comment:
No need to change, but we have a helper for this in `tests.common`: `async_mock_service` and `mock_service`